koku/koku/middleware.py

```diff
@@ -187,17 +190,15 @@ def _check_user_has_access(self, request):
             PermissionDenied: If the user does not have permissions for Cost Management.
 
         """
-
-        username = request.user.username
         if not request.user.admin and request.user.access is None:
-            msg = f"User {username} does not have permissions for Cost Management."
+            msg = f"User {request.user.username} does not have permissions for Cost Management."
             LOG.warning(msg)
             # For /user-access we do not want to raise the exception since the API will
             # return a false boolean response that the platfrom frontend code is expecting.
             if request.path != reverse("user-access"):
                 raise PermissionDenied(msg)
 
-    def _get_or_create_tenant(self, request):
+    def _get_tenant(self, request):
```

Comment:
This is a bit of a misnomer since this method does create a tenant in some circumstances. If it makes sense to change the name, the docstring needs to be updated as well.

Reply:
Can you name all the instances where we would be creating the public schema?